src/blob.rs

```diff
@@ -19,68 +28,154 @@ struct Header {
     flags: u64,
 }
 
-impl<K> Blob<K>
-where
-    K: Default,
-{
-    /// # Description
-    /// Create new blob from file
-    // @TODO more useful result
-    pub fn from_file<P: AsRef<Path>>(_path: P) -> Result<Self, ()> {
-        // @TODO implement
-        Ok(Self::default())
+#[derive(Debug)]
+pub struct WriteFuture {
+    file: fs::File,
+    buf: Option<Vec<u8>>,
+    current_offset: u64,
+}
+
+impl Future for WriteFuture {
+    type Output = Result<()>;
+
+    fn poll(mut self: Pin<&mut Self>, _waker: &Waker) -> Poll<Self::Output> {
+        let buf = self.buf.take().unwrap();
+        self.file.write_at(&buf, self.current_offset).unwrap();
```

Comment:
this is bad as it blocking. we can use waker to monitor fd. I have created #10

Reply:
In fahrenheit implemented only AsyncTcpStream/AsyncTcpListener. So I have to write implementation for files by myself. Looks like the new pull request